lib/Dialect/FIRRTL/FIRRTLFolds.cpp

```diff
@@ -2056,6 +2056,55 @@ static bool isPortUnused(Value port, StringRef data) {
   return true;
 }
 
+// Remove accesses to a port which is used.
+static void erasePort(PatternRewriter &rewriter, Value port) {
+  // Find the clock field of the port and determine whether the port is
+  // accessed only through its subfields or as a whole wire.  If the port
+  // is used in its entirety, replace it with a wire.  Otherwise,
+  // eliminate individual subfields and replace with reasonable defaults.
+  auto clkIndex = port.getType().cast<BundleType>().getElementIndex("clk");
+  Value clock;
+  for (auto *op : port.getUsers()) {
+    auto subfield = dyn_cast<SubfieldOp>(op);
+    if (!subfield) {
+      auto dummyWire = rewriter.create<WireOp>(port.getLoc(), port.getType());
```

Comment:
The old code was probably buggy and this inherits the bug, namely that wires must have a writer per the spec.  We are haphazard about enforcing this in transforms.  A register might be a better choice.

Reply:
(the clock could be a constant 0)

Reply:
Done.